spec/unit/query_azure_mock.rb

```diff
@@ -7,10 +8,31 @@ def setup_query_azure_mock
     stub_query_azure (@connection)
   end
 
+  def mock_list_images
+    create_service
+    stub_list_images(@service)
+  end
+
   def create_connection
     @connection = Azure::Connection.new(TEST_PARAMS)
   end
 
+  def create_service
+    @service = Azure::ServiceManagement::ASMInterface.new(TEST_PARAMS)
+  end
+
+  def stub_list_images(service)
+    stub_query_azure(service.connection)
+  end
+
+  def stub_delete_server(service)
+    stub_query_azure(service.connection)
+  end
+
+  def stub_show_server(service)
+    stub_query_azure(service.connection)
+  end
```

Comment:
@Vasu1105 , since there are more than one methods with the same definition, we should find some way out. For readability purpose, may be we can have a single method with some meaningful name and call the same method in all specs.